lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -259,9 +261,16 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
     rewriter.create<calyx::AssignOp>(loc, reg.getIn(), out);
     // The write enable port is high when the pipeline is done.
     rewriter.create<calyx::AssignOp>(loc, reg.getWriteEn(), opPipe.getDone());
+
+    // Set pipelineOp to hi as long as its done signal is not high.
+    // This prevents the pipelineOP from executing when we write into the
+    // register.
+    // To get !(pipelineOp.done) we do 1 xor pipelineOp.done
+    hw::ConstantOp c1 = createConstant(loc, rewriter, getComponent(), 1, 1);
     rewriter.create<calyx::AssignOp>(
-        loc, opPipe.getGo(),
-        createConstant(loc, rewriter, getComponent(), 1, 1));
+        loc, opPipe.getGo(), c1,
+        builder.create<comb::XorOp>(group.getLoc(), opPipe.getDone(), c1,
```

Comment:
CIRCT has a helper for creating a "not" using an XOR: https://github.com/llvm/circt/blob/72672984a7fa4092d2295a010527d647c2893ae3/lib/Dialect/Comb/CombOps.cpp#L48